tools/c7n_azure/c7n_azure/arm.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2018 Capital One Services, LLC
```

Comment:
Thoughts on creating an ArmResourceType in here too that covers good defaults for report field and name/id fields?  Most of that seems to be duplicated in each resource.

We could probably make this thing queryable if we wanted, so you could use it in a policy to act on generic resources? (useful for stuff like tagging and permissions at least)